sdk/src/trace/tracer_provider.cc

```diff
@@ -6,7 +6,16 @@ namespace sdk
 namespace trace
 {
 TracerProvider::TracerProvider(std::shared_ptr<SpanProcessor> processor) noexcept
-    : processor_{processor}, tracer_(new Tracer(std::move(processor)))
+    : processor_{processor},
```

Comment:
This can delegate to the other constructor:

```
TracerProvider::TracerProvider(std::shared_ptr<SpanProcessor> processor) noexcept 
    : TraceProvider(processor, std::make_shared<AlwaysOnSampler>()) {}
```